src/sentry/static/sentry/app/components/bases/pluginComponentBase.jsx

```diff
@@ -46,7 +46,7 @@ class PluginComponentBase extends React.Component {
     };
   }
 
-  UNSAFE_componentWillMount() {
+  componentDidMount() {
     this.api = new Client();
```

Comment:
This seems like something that should be happening in the constructor.

Reply:
Yeah agreed. Or maybe we can just wrap this whole thing with `withApi` and be done with it

Reply:
Subclasses use the class prop though

Reply:
ahh I see, nvm